grafeas-v1/lib/grafeas/v1/grafeas/client.rb

```diff
@@ -0,0 +1,1459 @@
+
+            default_config.rpcs.update_occurrence.timeout = 30.0
+
+            default_config.rpcs.get_occurrence_note.timeout = 30.0
+            default_config.rpcs.get_occurrence_note.retry_policy = {
+              initial_delay: 0.1,
+              max_delay:     60.0,
+              multiplier:    1.3,
+              retry_codes:   ["UNAVAILABLE", "DEADLINE_EXCEEDED"]
+            }
+
+            default_config.rpcs.get_note.timeout = 30.0
+            default_config.rpcs.get_note.retry_policy = {
+              initial_delay: 0.1,
+              max_delay:     60.0,
+              multiplier:    1.3,
+              retry_codes:   ["UNAVAILABLE", "DEADLINE_EXCEEDED"]
+            }
+
+            default_config.rpcs.list_notes.timeout = 30.0
+            default_config.rpcs.list_notes.retry_policy = {
+              initial_delay: 0.1,
+              max_delay:     60.0,
+              multiplier:    1.3,
+              retry_codes:   ["UNAVAILABLE", "DEADLINE_EXCEEDED"]
+            }
+
+            default_config.rpcs.delete_note.timeout = 30.0
+            default_config.rpcs.delete_note.retry_policy = {
+              initial_delay: 0.1,
+              max_delay:     60.0,
+              multiplier:    1.3,
+              retry_codes:   ["UNAVAILABLE", "DEADLINE_EXCEEDED"]
+            }
+
+            default_config.rpcs.create_note.timeout = 30.0
+
+            default_config.rpcs.batch_create_notes.timeout = 30.0
+
+            default_config.rpcs.update_note.timeout = 30.0
+
+            default_config.rpcs.list_note_occurrences.timeout = 30.0
+            default_config.rpcs.list_note_occurrences.retry_policy = {
+              initial_delay: 0.1,
+              max_delay:     60.0,
+              multiplier:    1.3,
+              retry_codes:   ["UNAVAILABLE", "DEADLINE_EXCEEDED"]
+            }
+
+            default_config
+          end
+          yield @configure if block_given?
+          @configure
+        end
+
+        ##
+        # Configure the Grafeas Client instance.
+        #
+        # The configuration is set to the derived mode, meaning that values can be changed,
+        # but structural changes (adding new fields, etc.) are not allowed. Structural changes
+        # should be made on {Client.configure}.
+        #
+        # See {::Grafeas::V1::Grafeas::Client::Configuration}
+        # for a description of the configuration fields.
+        #
+        # @yield [config] Configure the Client client.
+        # @yieldparam config [Client::Configuration]
+        #
+        # @return [Client::Configuration]
+        #
+        def configure
+          yield @config if block_given?
+          @config
+        end
+
+        ##
+        # Create a new Grafeas client object.
+        #
+        # ## Examples
+        #
+        # To create a new Grafeas client with the default
+        # configuration:
+        #
+        #     client = ::Grafeas::V1::Grafeas::Client.new
```

Comment:
Does this example still work without default credentials?

Reply:
It might, if the underlying implementation doesn't require credentials or has a different default credentialing mechanism than Google's. This is a generic library and we have to omit any details that could be different for different implementations.